File: frontend/src/types/financeflowDashboard.ts
export interface FinanceFlowDashboard {
  capital_added: string;
  available_capital: string;
  capital_currently_lent: string;
  principal_outstanding: string;
  profit_today: string;
  profit_this_month: string;
  total_profit: string;
  available_profit: string;
  active_loans: number;
  completed_loans: number;
  overdue_loans: number;
  total_borrowers: number;
  expected_today: string;
  collected_today: string;
  pending_today: string;
  collection_rate: string;
  unsettled_with_agents: string;
  pending_settlement_count: number;
  pending_settlement_total: string;
}

File: backend/app/schemas/financeflow_dashboard.py
from decimal import Decimal

from pydantic import BaseModel


class FinanceFlowDashboardResponse(BaseModel):
    capital_added: Decimal
    available_capital: Decimal
    capital_currently_lent: Decimal
    principal_outstanding: Decimal
    profit_today: Decimal
    profit_this_month: Decimal
    total_profit: Decimal
    available_profit: Decimal
    active_loans: int
    completed_loans: int
    overdue_loans: int
    total_borrowers: int
    expected_today: Decimal
    collected_today: Decimal
    pending_today: Decimal
    collection_rate: Decimal
    unsettled_with_agents: Decimal = Decimal("0")
    pending_settlement_count: int = 0
    pending_settlement_total: Decimal = Decimal("0")
